frontend/src/v4/routes/viewerGui/components/activities/components/activityDetails/activityDetails.component.tsx

```diff
@@ -49,7 +49,10 @@ const renderValue = (key, value, setSelectedDate) => {
 	}
 
 	return (
-		<Grid item xs={6}>{processedValue} <TimeIcon name={dateToSelect ? undefined : key} value={dateToSelect || value} handleOnClick={setSelectedDate} /></Grid>
+		<Grid item xs={6}>
+			{processedValue}
+			<TimeIcon name={dateToSelect ? undefined : key} handleOnClick={() => setSelectedDate(dateToSelect || value)} />
```

Comment:
What got fixed by this?

Reply:
Nothing really, as I was looking for the component responsible for fixing the problem I was fixing, I thought this could have been it, so I refactored it for readability. I'm going to revert this to keep the PR as clear as possible